exec/java-exec/src/main/java/org/apache/drill/exec/resolver/DefaultFunctionResolver.java

```diff
@@ -38,12 +38,11 @@ public DrillFuncHolder getBestMatch(List<DrillFuncHolder> methods, FunctionCall
     int currcost = Integer.MAX_VALUE;
     DrillFuncHolder bestmatch = null;
     final List<DrillFuncHolder> bestMatchAlternatives = new LinkedList<>();
-
+    List<TypeProtos.MajorType> argumentTypes = Lists.newArrayList();
```

Comment:
```
    List<TypeProtos.MajorType> argumentTypes = call.args().stream()
      .map(LogicalExpression::getMajorType)
      .collect(Collectors.toList());
```

Reply:
ok, but I think we should open to contributors to choose the function programing or command one.

Reply:
Well, it's just a suggestion... it's looks much better.
If you want to leave as it you welcome but than `Lists.newArrayList();` replace Guave usage then.

Reply:
got it. I have applied your advice. thanks @arina-ielchiieva